### PR TITLE
HOTFIX: fix ProcessorStateManager to use correct ktable partitions

### PR DESCRIPTION
**Pull request by ymatsuda**
@guozhangwang 
- fix ProcessorStateManager to use correct ktable partitions
- more ktable tests

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StandbyTaskTest.java [hunk 1]
```diff
@@ -66,6 +69,15 @@
             )
     );
 
+    private final TopicPartition ktable = new TopicPartition("ktable1", 0);
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StandbyTaskTest.java:78]: nit: renamet to "ktablePartition"

**Comment by guozhangwang** [comment]
LGTM overall.

**Comment by guozhangwang** [comment]
@ymatsuda Could you rebase this PR and add the test case for KTable stores to make sure the topic name is correctly constructed by the restore consumer?

**Comment by ymatsuda** [comment]
The recent changes are pulled in.

**Comment by guozhangwang** [comment]
Merged to trunk.

**Pull request closed without merging**